drivers/gpio/gpio_nrfx.c

```diff
@@ -368,18 +422,59 @@ static uint32_t check_level_trigger_pins(const struct device *port)
 	uint32_t pin = 0U;
 	uint32_t bit = 1U << pin;
 
-	while (level_pins) {
-		if (level_pins & bit) {
+	uint32_t port_latch = 0;
+
+#if IS_ENABLED(CONFIG_GPIO_NRF_INT_EDGE_USING_SENSE)
+	uint32_t int_pins = data->pin_int_en;
+
+	/* Read LATCH, which will tell us which pin has changed its state. */
+	port_latch = cfg->port->LATCH;
```

Comment:
If the input signal for a latched detection changes state again before the level is reconfigured below, the configured state will be inconsistent with the signal state and the next change will not be detected.

Reply:
I am not sure what the problem is, so let me describe how I understand what happens in following cases:

1) simple case with one edge
```
DETECT 0 1|0|0|0
LATCH  0 1|0|0|0
SENSE  1 1|.|0|0
STATE  0 1|1|1|1
          ^ ^ ^
          | | |
          | | fire callback
          | |
          | reconfigure SENSE
          |
  disable SENSE and clear LATCH
```
Here we start with `0` (low) signal on GPIO pin, which is described above in `STATE` row. `SENSE` is configured to detect the opposite signal, which is `1` (high). When `STATE` changes, then `LATCH` (per pin) and `DETECT` (per port) are set. This is when ISR starts and `check_level_trigger_pins()` is executed. This function is responsible for "disable SENSE and clear LATCH". Disabled `SENSE` is described as `.` above. Then `SENSE` is reconfigured to the opposite signal, which is now `0` (low). At the end gpio callback is fired and ISR exits.

2) one edge, followed by second edge before reconfiguring `SENSE`

```
DETECT 0 1|0|1|1
LATCH  0 1|0|1|1
SENSE  1 1|.|1|1
STATE  0 1|0|0|0
          ^ ^ ^
          | | |
          | | fire callback
          | |
          | reconfigure SENSE
          |
  disable SENSE and clear LATCH
```
GPIO pin state is changed just after `SENSE` is disabled and `LATCH` was cleared. This state change is not "detected" immediately. Then `SENSE` is reconfigured to the opposite state, which automatically results in `LATCH` and `DETECT` being set, as this already matches state on GPIO pin. At the end gpio callback is fired and ISR exits. After that ISR is reentered once again, as `DETECT` is high.

3) one edge, followed by two edges before reconfiguring `SENSE`
```
DETECT 0 1|0 0|0|0
LATCH  0 1|0 0|0|0
SENSE  1 1|. .|0|0
STATE  0 1|0 1|1|1
          ^   ^ ^
          |   | |
          |   | fire callback
          |   |
          |   reconfigure SENSE
          |
  disable SENSE and clear LATCH
```
GPIO pin state is changed (`1` -> `0`) just after `SENSE` is disabled and `LATCH` was cleared. Then it changes once again (`0` -> `1`). Those state changes are not "detected" at this point. Then `SENSE` is reconfigured to the opposite state, which is `0` (low). Later, gpio callback is fired and ISR exits. ISR is **not** reentered, which is fine, because there were no edges after gpio callback was fired.

Reply:
OK, thanks.  I understand better about how LATCH plus level sense avoids the concerns I had.  In my use case it's more important to be able to detect (actually, count) edge transitions more accurately, but the Zephyr GPIO API doesn't support that anyway so it's irrelevant.

I guess this probably works, though I'd put more comments into the code explaining exactly how all those conditions guarantee things work.  It really isn't obvious.

I still think the preferred path is one that adds per-instance configuration, by which I mean a specific pin that requires edge triggers should be allowed to select between GPIOTE-detected and level-detected solutions.  But this isn't something Zephyr's prepared to support.

So, for my review the current state is just get rid of the `check_pins` alias and my input is addressed.